packages/miniflare/src/plugins/assets/index.ts

```diff
@@ -196,9 +212,26 @@ const walk = async (dir: string) => {
 					);
 				}
 
-				manifest.push(
-					await hashPath(encodeFilePath(relativeFilepath, path.sep))
+				// Each manifest entry is [header, pathHash, contentHash].
+				// In dev we want to avoid reading and hashing every file every time we
+				// reload the dev server, so in dev the 'content hash' is hash(path + modifiedTime).
+				// This way the 'content hash' still updates on file change, so the eTag also updates,
+				// and we don't incorrectly get a 304.
+				const pathHash = await hashPath(
+					encodeFilePath(relativeFilepath, path.sep)
+				);
+				const modifiedTime = (await fs.stat(filepath)).mtimeMs;
+				const contentHash = await hashPath(
+					encodeFilePath(relativeFilepath, path.sep) + modifiedTime.toString()
 				);
```

Comment:
Nit: if you can reuse the `filestat` above, you can parallelize the two hashPath calls, e.g.
```ts
const [pathHash, contentHash] = await Promise.all(
  hashPath(encodeFilePath(relativeFilepath, path.sep)),
  hashPath(encodeFilePath(relativeFilepath, path.sep) + filestate.mtimeMs.toString())
);
```